pipeline_dp/dataframes.py

```diff
@@ -0,0 +1,189 @@
+        Args:
+            column: column to group.
+            max_groups_contributed: the maximum groups that can each privacy
+              unit contributes to the result. If some privacy unit contributes
+              more in the input dataset, the groups are sub-sampled to
+              max_groups_contributed.
+            max_contributions_per_group: the maximum contribution that a privacy
+              unit can contribute to a group. If some privacy unit contributes
+              more to some group, contributions are sub-sampled to
+              max_contributions_per_group.
+            public_keys:
+        """
+        if self._groupby_column is not None:
+            raise ValueError("groupby can be called only once.")
+        if column not in self._df.columns:
+            raise ValueError(f"Column {column} is not present in DataFrame")
+        self._groupby_column = column
+        self._contribution_bounds.max_partitions_contributed = max_groups_contributed
+        self._contribution_bounds.max_contributions_per_partition = max_contributions_per_group
+        self._public_keys = public_keys
+        return self
+
+    def count(self, name: str = None) -> 'QueryBuilder':
+        """Adds count to the query.
+
+        Args:
+            name: the name of the output column.
+        """
+        if self._groupby_column is None:
+            raise ValueError(
+                "Global aggregations are not supported. Use groupby.")
+        if pipeline_dp.Metrics.COUNT in self._metrics:
+            raise ValueError("count can be counted only once.")
+        self._metrics[pipeline_dp.Metrics.COUNT] = name
+        return self
+
+    def sum(self,
+            column: str,
+            *,
+            min_value: float,
+            max_value: float,
+            name: str = None) -> 'QueryBuilder':
+        """Adds sum to the query.
+
+        Args:
+            column: column to perform summation
+            min_value, max_value: capping limits to each value.
+            name: the name of the output column.
+        """
+        if self._groupby_column is None:
+            raise ValueError(
+                "Global aggregations are not supported. Use groupby.")
+        if pipeline_dp.Metrics.SUM in self._metrics:
+            raise ValueError("sum can be counted only once.")
+        if self._value_column is not None:
+            raise NotImplementedError(
+                "Now aggregation of only 1 column is supported.")
+        self._metrics[pipeline_dp.Metrics.SUM] = name
+        self._value_column = column
+        self._contribution_bounds.min_value = min_value
+        self._contribution_bounds.max_value = max_value
+        return self
+
+    def build_query(self) -> Query:
+        """Builds the DP query."""
+        if self._groupby_column is None:
+            raise NotImplementedError(
+                "Global aggregations are not implemented yet. Call groupby.")
+        metrics = list(self._metrics.keys())
```

Comment:
should we assert that metrics is not empty?

also, Query expects dictionary and we take here keys only

Reply:
Thx, good point. I've added check

Reply:
but we still get keys from metrics and pass to Query metrics as list. But the declartion of Query consturctor says that metrics is Dict[Metric, List[str]]. List is strange too. I think it should be Dict[Metric, str] and we don't need to do anything with self._metrics, just pass it as it is.

Reply:
Thanks, done